src/main/java/umaikaze/duke/task/Task.java

```diff
@@ -0,0 +1,26 @@
+package umaikaze.duke.task;
+
+import umaikaze.duke.DukeException;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public void markDone() {
+        isDone = true;
+    }
+
+    public String getSaveString() throws DukeException {
```

Comment:
As mentioned above, you might consider implementing the `getSaveString()` method by extracting the common parts of the string for different types of tasks.